types/foundry/common/prosemirror/_module.d.mts

```diff
@@ -23,23 +23,23 @@ import { parseHTMLString, serializeHTMLString } from "./util.mjs";
 declare const dom: {
     parser: DOMParser;
     serializer: DOMSerializer;
-    parseString: parseHTMLString;
-    serializeString: serializeHTMLString;
+    parseString: typeof parseHTMLString;
+    serializeString: typeof serializeHTMLString;
 };
 
 declare const defaultPlugins: {
-    inputRules: ProseMirror.Plugin;
-    keyMaps: ProseMirror.Plugin;
-    menu: ProseMirror.Plugin;
-    isDirty: ProseMirror.Plugin;
-    clickHandler: ProseMirror.Plugin;
-    pasteTransformer: ProseMirror.Plugin;
-    baseKeyMap: ProseMirror.Plugin;
-    dropCursor: ProseMirror.Plugin;
-    gapCursor: ProseMirror.Plugin;
-    history: ProseMirror.Plugin;
-    columnResizing: ProseMirror.Plugin;
-    tables: ProseMirror.Plugin;
+    inputRules: Plugin;
+    keyMaps: Plugin;
+    menu: Plugin;
+    isDirty: Plugin;
+    clickHandler: Plugin;
+    pasteTransformer: Plugin;
+    baseKeyMap: Plugin;
+    dropCursor: Plugin;
+    gapCursor: Plugin;
+    history: Plugin;
+    columnResizing: Plugin;
+    tables: Plugin;
 };
```

Comment:
Are these prosemirror plugins or foundry plugin classes? Does the foundry one implement the prosemirror one?

Reply:
`ProseMirrorPlugin` is not an extension of `ProseMirror.Plugin`. The static `create` method builds a `ProseMirror.Plugin` and returns `ProseMirrorPlugin#render` (html) for the editor view.

`ProseMirrorMenu` looks like this:
```ts
  static build(schema, options={}) {
    return new Plugin({
      view: editorView => {
        return new this(schema, editorView, options).render();
      }
    });
  }
```